common/app/model/CrosswordData.scala

```diff
@@ -81,7 +81,12 @@ object CrosswordData {
     // group are incorrectly stored on the first group entry. We normalize
     // the data to store the separator locations on their corresponding entries.
 
-    val entries = crossword.entries.map(Entry.fromCrosswordEntry)
+    val shipSolutions = crossword.dateSolutionAvailable.map(_.toJodaDateTime.isBeforeNow).getOrElse(crossword.solutionAvailable)
+
+    val entries = crossword.entries.collect {
+      case entry if shipSolutions || entry.solution.isEmpty => Entry.fromCrosswordEntry(entry)
+      case entry => Entry.fromCrosswordEntry(entry.copy(solution = None))
```

Comment:
This may be slightly easier to read if you invert the check:

```scala
val entries = crossword.entries.collect {
  case entry if !shipSolutions => Entry.fromCrosswordEntry(entry.copy(solution = None))
  case entry => Entry.fromCrosswordEntry(entry)
}
```